util/store/store.go

```diff
@@ -138,14 +132,9 @@ func SetInt(store sdk.KVStore, key []byte, val sdkmath.Int, errField string) err
 func GetDec(store sdk.KVStore, key []byte, errField string) sdk.Dec {
 	val := GetValue[*sdk.Dec](store, key, errField)
 	if val == nil {
-		// No stored bytes at key: return zero
+		// Not found: return zero
 		return sdk.ZeroDec()
 	}
-
-	if val.IsNegative() {
-		panic(fmt.Sprintf("%s: retrieved negative %s", val, errField))
-	}
```

Comment:
this check is already done in `SetDec` , so not needed.
In fact, we should do validity check in the "layer" above.

Reply:
The philosophy behind checking for negative/invalid value in getters was explained in the comment at the top of the file:

Two main reasons
- Edge cases. Basically, even if we botch a migration, or load a store that was made with different Set methods (e.g. existing modules), or if some bad code doesn't use `SetDec` and instead stores a negative dec manually, this function will detect it. This is further enhanced by `errField`: if we run into this error unexpectedly, the cause will always be known in the error message (even if it's reported by a user or in a context where we don't have stack trace)
- Simplifies higher layers. If we check for negative/nil in this one file, all modules above it do not have to reproduce the same validation logic after every time they use `GetDec` It also makes it impossible to forget to include that validation.

Reply:
I do see your point about not making assumptions in store - I am not sure if there is any context in our existing modules where a store negative number would be correct. It's a tradeoff between purity (no assumptions) and standardization (second point above)

Reply:
Validation must happen on value operations (input, request processing, arithmetics), not on the store value, where we are out of the context.